site/content/docs/cli/constructs.md

```diff
@@ -0,0 +1,422 @@
+```
+2. Every `logicalId` needs to be unique within the scope of a `Project`. A Project also has a `logicalId`.
+3. A `logicalId` can be any string up to 255 characters in length.
+4. There is no hard limit on the amount of `Project`'s you can have in your Checkly account.
+
+Behind the scenes, we use the `logicalId` to create a graph of your resources so we know what to persist, update and remove
+from our database. Changing the `logicalId` on an existing resource in your code base will tell the Checkly backend that
+a resource was removed and a new resource was created.
+
+So, I guess you know now that logical IDs are important!
+
+## Creating an API Check
+
+API checks are used to validate your HTTP based API endpoints. Let's look at the example below as it does a couple of things:
+
+- It defines the basic check properties like `name`, `activated` etc.
+- It defines the HTTP method `GET` the `url`.
+- It defines an array of assertions to assert the HTTP response status is correct.
+
+```js
+// hello-api.check.js
+
+const { ApiCheck } = require('@checkly/cli/constructs')
+const path = require('path')
+const { readFileSync } = require('fs')
+
+new ApiCheck('hello-api-1', {
+  name: 'Hello API',
+  activated: true,
+  request: {
+    method: 'GET',
+    url: 'https://mac-demo-repo.vercel.app/api/hello',
+    assertions: [
+      { source: 'STATUS_CODE', regex: '', property: '', comparison: 'EQUALS', target: '200' },
+    ]
+  }
+})
+```
+
+## Creating and adding an Alert Channel
+
+When a check fails, you want to get alerted. There are two steps to take here:
+
+1. Create one or more alert channels. You can put them in a different file to DRY up your code, i.e. in `alert-channels.js`
+
+```js
+// alert-channels.js
+
+const { SmsAlertChannel, EmailAlertChannel } = require('@checkly/cli/constructs')
+
+const sendDefaults = {
+  sendFailure: true,
+  sendRecovery: true,
+  sendDegraded: false,
+}
+
+const smsChannel = new SmsAlertChannel('sms-channel-1', {
+  phoneNumber: '0031061234567890',
+  ...sendDefaults
+})
+
+const emailChannel = new EmailAlertChannel('email-channel-1', {
+  address: 'alerts@acme.com',
+  ...sendDefaults
+})
+
+module.exports = {
+  smsChannel,
+  emailChannel
+}
+```
+
+2. Now you can import these channels into one or more checks by passing the objects into the `alertChannels` array:
+
+```js
+// api.check.js
+
+const { ApiCheck } = require('@checkly/cli/constructs')
+const { smsChannel, emailChannel } = require('./alert-channels')
+
+new ApiCheck('hello-api-1', {
+  name: 'Hello API',
+  alertChannels: [smsChannel, emailChannel],
+  request: {
+    method: 'GET',
+    url: 'https://mac-demo-repo.vercel.app/api/hello',
+  }
+})
+```
+
+# Using the Constructs API
+
+All resources you can create and manage using the Checkly CLI are derived from "constructs". These constructs are just
+[TypeScript classes](https://github.com/checkly/checkly-cli/tree/main/package/src/constructs) like `ApiCheck` in `api-check.ts` and
```

Comment:
This feels like repetition from the into. Should TypeScript be mentioned upfront?

Reply:
I don't want to assume people read the docs from top to bottom. People will enter pages through Google, or by other references in the UI or terminal. Some repetition is therefore OK.

Reply:
But shouldn't we then still also include the mention and link to the TS classes at the top? 

The link is a 404 btw. :)

Reply:
fixed the link